client/app/components/InputWithCopy.jsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import Input from 'antd/lib/input';
+import Icon from 'antd/lib/icon';
+import Tooltip from 'antd/lib/tooltip';
+
+export default class InputWithCopy extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { copied: null };
+    this.ref = React.createRef();
+    this.copyFeatureSupported = document.queryCommandSupported('copy');
+    this.resetCopyState = null;
+  }
+
+  componentWillUnmount() {
+    if (this.resetCopyState) {
+      clearTimeout(this.resetCopyState);
+    }
+  }
+
+  copy = () => {
+    // select text
+    this.ref.current.select();
+
+    // copy
+    try {
+      const success = document.execCommand('copy');
+      if (!success) {
+        throw new Error();
```

Comment:
Why try/catch? :-)

```javascript
const success = document.execCommand('copy');
this.setState({ copied: success ? 'Copied!' : 'Copy failed' });
```

Reply:
My guessing was that `document.execCommand()` could throw Errors on its own, but I'm not sure and I didn't find ath related :thinking:

Reply:
[According to docs](https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand) it does not throw errors, just returns `boolean` value (which also is meaningless if `document.execCommand()` not called as a part of user interaction, but in this case it's ok)

Reply:
Yeah a `SecurityError` https://dvcs.w3.org/hg/editing/raw-file/tip/editing.html#the-copy-command

Reply:
Hm. There is some conflict in that docs:

![image](https://user-images.githubusercontent.com/12139186/52640063-55495f00-2ede-11e9-8fa9-74e06b6e7bb2.png)

and then:

![image](https://user-images.githubusercontent.com/12139186/52640099-672b0200-2ede-11e9-9c48-d2984c8f5e7a.png)

Cant say who tells the truth, so let's keep try/catch block.